### PR TITLE
[WIP][hyperapi 7.0.9] Adding tsforecaster create model  [HyperAPI] 

### PR DESCRIPTION
**Pull request by WassimAbida**

### CODE REVIEW

File: HyperAPI/hyper_api/model.py [hunk 1]
```diff
@@ -782,6 +785,130 @@ def create_Perceptron(self, dataset, name, target, penalty=None, alpha=1e-4,fit_
...
+                                    scaler_target='', scaler_X_var='', grid_search=False, rolling_enrich_features=[], test_size=0.3, regressor={},
+                                    enable_custom_discretizations=True, nbMaxModality=50, nbMinObservation=10, replaceMissingValues='Median'):
+        """
+        Create a eXtreme Gradient Boosting (XGBoost) regressor
+
+        Args:
+            dataset (Dataset): Dataset the model is fitted on
+            name (str): Name of the new model
+            target (Target): Target used to generate the model
+            target_name (str): Name of the target
```
Comment (AlainSenotH3) [inline HyperAPI/hyper_api/model.py]: ? this is not an independant input parameter

File: HyperAPI/hyper_api/model.py [hunk 2]
```diff
@@ -782,6 +785,130 @@ def create_Perceptron(self, dataset, name, target, penalty=None, alpha=1e-4,fit_
...
+            one_hot_encode_seasons (bool): One hot encode the seasonalities [categorical features]
+            remove_seasons (bool): remove seasonalities for features
+            dropna (bool): remove NaNs values
+            scaler_target (str): name of scaler for time series
+            scaler_X_var (str): name of scaler for X values (features)
+            grid_search (bool): apply grid search for XGboost Regressor
+            rolling_enrich_features (list): features to include additional rolling statistics
+            test_size (float): test size 
+            regressor (dict): defines the regressor algorithm to use and its hyperparameters
+
```
Comment (AlainSenotH3) [inline HyperAPI/hyper_api/model.py]: some args are missing

File: HyperAPI/hyper_api/model.py [hunk 3]
```diff
@@ -782,6 +785,130 @@ def create_Perceptron(self, dataset, name, target, penalty=None, alpha=1e-4,fit_
...
+                    "loss": "mae", "metrics": "mse", "optimizer_name": "sgd",
+                    "optimizer_params": {"lr": 0.01, "decay": 1e-6, "momentum": 0.9, "nesterov": true},
+                    "callbacks": {"monitor": "val_loss", "mode": "min", "patience": 5},
+                    "epochs": 100, "batch_size": 128, "validation_split": 0.3,
+                    "shuffle": false, "verbose": 0}
+                                    }
+        Returns:
+            the created model
+        """
+        regressor = {
```
Comment (AlainSenotH3) [inline HyperAPI/hyper_api/model.py]: isn't it overriding the parameter from the user ?

File: HyperAPI/hyper_api/model.py [hunk 4]
```diff
@@ -782,6 +785,130 @@ def create_Perceptron(self, dataset, name, target, penalty=None, alpha=1e-4,fit_
...
+            n_lags_max (int): maximum lags to include
+            lags_to_include (list): specific lags to include
+            one_hot_encode_seasons (bool): One hot encode the seasonalities [categorical features]
+            remove_seasons (bool): remove seasonalities for features
+            dropna (bool): remove NaNs values
+            scaler_target (str): name of scaler for time series
+            scaler_X_var (str): name of scaler for X values (features)
+            grid_search (bool): apply grid search for XGboost Regressor
+            rolling_enrich_features (list): features to include additional rolling statistics
+            test_size (float): test size
```
Comment (AlainSenotH3) [inline HyperAPI/hyper_api/model.py]: is it the same parameter as the one usually containing 'ratio' in its name ?

File: HyperAPI/hyper_api/model.py (hunk 1 above)
Comment (AlainSenotH3) [reply HyperAPI/hyper_api/model.py]: isn't it target.variable_name ?

File: HyperAPI/hyper_api/model.py [hunk 5]
```diff
@@ -782,6 +784,145 @@ def create_Perceptron(self, dataset, name, target, penalty=None, alpha=1e-4,fit_
...
+            nbMinObservation (int): Modalities with a number of observations lower will be ignored. Default is 10
+            replaceMissingValues (str): Method to replace missing values. Available methods are 'Median', 'Mean' and 'Delete'. Default is 'Median'
+            
+        Returns:
+            the created model
+        """
+        # case empty regressor entry
+        if regressor == {}:
+            import logging
+            return logging.info('Give a valid regresso choice')
```
Comment (AlainSenotH3) [inline HyperAPI/hyper_api/model.py]: you should not return this

**Comment by JeromeMaslowski** [comment]
Wait until 7.0.7 is merged.

File: changelog.md [hunk 6]
```diff
@@ -2,6 +2,9 @@
 
 ## Version 7
 
+### 7.0.9 
+- Adding Time Series Forecasting Model
+
 ### 7.0.8
 - Using string delimiters when creating datasets in mongo
```
Comment (JeromeMaslowski) [inline changelog.md]: 7.0.7 is missing. This PR will need to be rebased.

**Comment by serkansahin** [comment]
duplicate of https://github.com/HyperCube/HyperAPI/pull/127 closing

**Pull request closed without merging**